src/main/java/com/google/sps/servlets/GroceryServlet.java

```diff
@@ -27,39 +27,108 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
+// TODO: fix git error when pushing
 /** Servlet takes grocery list and displays it to the screen based on the users family */
 @WebServlet("/grocery-list")
 public class GroceryServlet extends HttpServlet {
+  private static final String GROCERY = "Grocery";
+  private static final String FAMILY_ID = "familyID";
+
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {      
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String userEmail = userService.getCurrentUser().getEmail();
 
-    // TODO: add items to the specific families datastore.   
+    // if user isn't associated with family don't allow them to create a list
+    Query query = new Query("UserInfo")
+        .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, userEmail));
+    PreparedQuery results = datastore.prepare(query);
+    Entity userInfoEntity = results.asSingleEntity();
+    if (userInfoEntity == null) {
+        response.sendRedirect("/grocery.html");
+        return;
+    }
+    String assignGrocery = request.getParameter("assignGrocery");
+    String noneAssigned = " ";
+    Entity groceryEntity = new Entity(GROCERY);
+   
+    // turn familyID into a key so the users family can be accessed
+    long familyID = (long) userInfoEntity.getProperty(FAMILY_ID);
+    Key familyEntityKey = KeyFactory.createKey("Family", familyID);
+    Entity familyEntity;
+    try {
+        familyEntity = datastore.get(familyEntityKey);
+    } catch (EntityNotFoundException e) {
+        System.out.println("Family not found");
+        return;
+    }
+    System.out.println("grocery assigned to: " + assignGrocery);
+    // checks if the given email matches a email in the family
+    if (assignGrocery.equals("") || assignGrocery.equals(null)) {
+        groceryEntity.setProperty("assignEmail", noneAssigned);
+    } else {
+        ArrayList<String> memberEmails = (ArrayList<String>) familyEntity.getProperty("memberEmails");
+        for(int i = 0; i < memberEmails.size(); i++) {
+            if(assignGrocery.equals(memberEmails.get(i))) {
+                groceryEntity.setProperty("assignEmail" , memberEmails.get(i));
```

Comment:
This essentially works out to be the following, right?

```
if memberEmails.contains(assignEmail):
  set the groceryEntity property
```

There's a contains function on ArrayList that will allow you to make that check, so no need for a loop.

Bonus points for changing the ArrayList to a HashSet. :)

But also, in what case are you expecting the user not to be part of the family? You fetched the family by looking up the familyId for the user, right? So in that case, I think we would know the family is the user's?